src/Reflex/Dynamic.hs

```diff
@@ -232,12 +235,21 @@ distributeListOverDynPure =
     fromDSum :: DSum (Const2 Int a) Identity -> a
     fromDSum (Const2 _ :=> Identity v) = v
 
---TODO: Generalize this to functors other than Maps
 -- | Combine a 'Dynamic' of a 'Map' of 'Dynamic's into a 'Dynamic'
 -- with the current values of the 'Dynamic's in a map.
 joinDynThroughMap :: forall t k a. (Reflex t, Ord k) => Dynamic t (Map k (Dynamic t a)) -> Dynamic t (Map k a)
 joinDynThroughMap = joinDyn . fmap distributeMapOverDynPure
 
+-- | Combine a 'Dynamic' of a 'Traversable' of 'Dynamic's into a 'Dynamic'
+-- with the current values of the 'Dynamic's in the functor.
+joinDynThroughTraversable :: (Reflex t, Traversable f) => Dynamic t (f (Dynamic t a)) -> Dynamic t (f a)
+joinDynThroughTraversable = join . fmap sequence
+
+-- | Combine a 'Dynamic' of a 'Distributive' of 'Dynamic's into a functor of 'Dynamic's.
+-- The current value at each hole in the resulting functor will be the current value of that hole in the current original functor
+joinDynThroughDistributive :: (Reflex t, Distributive f) => Dynamic t (f (Dynamic t a)) -> f (Dynamic t a)
+joinDynThroughDistributive = fmap join . distribute
```

Comment:
I believe `distribute` will make all the child `updated` events trigger whenever the parent one does. 
Is this what we want? Do we need `holdUniqDyn` in there?

Reply:
Can't do better without extra constraints.  In fact, not sure we should have this because almost 100% of the time it will be better to have the Dynamic be inside the f in the first place, *especially* for Distributive where there's nothing outside the 'holes' that can change ...